pkg/backend/filestate/state.go

```diff
@@ -260,7 +283,9 @@ func (b *localBackend) removeStack(name tokens.QName) error {
 func backupTarget(bucket Bucket, file string) string {
 	contract.Require(file != "", "file")
 	bck := file + ".bak"
-	err := renameObject(bucket, file, bck)
+	err := bucket.Delete(context.TODO(), bck)
```

Comment:
This is the part I least like about this PR, but I think it should not cause problems.
This `Delete` is here so we don't have stale `.json.bak` next to newer `.json.gz.bak` and inversely.

Reply:
But the renameObject below is going to write over bck anyway? This delete looks superfluous.